src/twisted/trial/_dist/worker.py

```diff
@@ -68,20 +79,44 @@ class WorkerProtocol(AMP):
     The worker-side trial distributed protocol.
     """
 
+    logger = Logger()
+
     def __init__(self, forceGarbageCollection=False):
         self._loader = TestLoader()
         self._result = WorkerReporter(self)
         self._forceGarbageCollection = forceGarbageCollection
 
     @workercommands.Run.responder
-    def run(self, testCase):
+    async def run(self, testCase: str) -> RunResult:
         """
         Run a test case by name.
         """
-        case = self._loader.loadByName(testCase)
-        suite = TrialSuite([case], self._forceGarbageCollection)
-        suite.run(self._result)
-        return {"success": True}
+        with self._result.gatherReportingResults() as results:
+            case = self._loader.loadByName(testCase)
+            suite = TrialSuite([case], self._forceGarbageCollection)
+            suite.run(self._result)
+
+            allSucceeded = True
+            for (success, result) in await DeferredList(results, consumeErrors=True):
+                if not success:
+                    allSucceeded = False
+                    # We can try to report the error but since something has
+                    # already gone wrong we shouldn't be extremely confident
+                    # that this will succeed.  So we will also log it and any
+                    # errors reporting it to our local log.
+                    self.logger.failure(
+                        "Result reporting for {id} failed",
+                        failure=result,
+                        id=testCase,
+                    )
+                    try:
+                        await self._result._addErrorFallible(testCase, result)
```

Comment:
Maybe have it a public interface

```suggestion
                        await self._result.addErrorFallible(testCase, result)
```

The idea is that maybe in the future we would want to run some code scanners to detect for private access.